src/shared/dissect-image.c

```diff
@@ -2004,10 +2004,25 @@ static int mount_partition(
 
         if (p) {
                 if (m->fsmount_fd >= 0) {
+                        bool mount_beneath = FLAGS_SET(flags, DISSECT_IMAGE_MOUNT_BENEATH);
+
                         /* Case #1: Attach existing fsmount fd to the file system */
 
-                        if (move_mount(m->fsmount_fd, "", -EBADF, p, MOVE_MOUNT_F_EMPTY_PATH) < 0)
-                                return -errno;
+                        r = RET_NERRNO(move_mount(m->fsmount_fd,
+                                       "",
+                                       -EBADF,
```

Comment:
if you touch this, maybe update to current coding style, i.e. `/* from_path= */ "",` and `/* to_fd= */`...